app/src/main/java/edu/uc/hickmadc/customnotifications/ui/main/MainViewModel.kt

```diff
@@ -1,21 +1,22 @@
 package edu.uc.hickmadc.customnotifications.ui.main
 
-import android.view.View
 import androidx.lifecycle.ViewModel
 import edu.uc.hickmadc.customnotifications.dto.Group
+import edu.uc.hickmadc.customnotifications.service.GroupService
 import java.util.logging.Logger
 
 class MainViewModel : ViewModel() {
+
+    var groups : ArrayList<Group> = ArrayList()
+    var groupService : GroupService = GroupService()
+
     fun exampleClick() {
         Logger.getAnonymousLogger().warning("You just pressed the add group button")
     }
 
-    fun getAllGroups() : ArrayList<Group> {
-        var groups : ArrayList<Group> = ArrayList()
-        //placeholder, should be replaced by database call/live data
-        groups.add(Group(name = "Group1", desc = "This data is", active = true, same_schedule = false, group_id = 1))
-        groups.add(Group(name = "Group2", desc = "set in the", active = false, same_schedule = false, group_id = 1))
-        groups.add(Group(name = "Group3", desc = "main view model", active = true, same_schedule = false, group_id = 2))
+    fun getAllGroups(): ArrayList<Group> {
+        groups = groupService.getAllGroups()
         return groups
+
```

Comment:
Suggestion 2: Moved this code down to the service layer, and added functionality for the "getAllGroups" function to pass down to that layer rather than do it all in the MainViewModel.
Source: https://docs.sonarqube.org/display/SONARQUBE45/Bad+Distribution+of+Complexity

Reply:
Great idea.  This declutters the ViewModel.